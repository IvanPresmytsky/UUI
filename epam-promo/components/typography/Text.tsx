import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type TextSize = 'none' | '18' | '24' | '30' | '36' | '42' | '48';

interface TextMods {
    /**
     * Defines component color.
     * @default 'gray80'
     */
    color?: 'blue' | 'green' | 'amber' | 'red' | 'white' | 'gray5' | 'gray50' | 'gray60' | 'gray80' | 'gray90' | uui.TextProps['color'];
    /**
     * Defines text size
     * @default '36'
     */
    size?: TextSize;
}

/** Represents the properties of a Text component. */
export interface TextProps extends uui.TextCoreProps, TextMods {}

export const Text = createSkinComponent<uui.TextProps, TextProps>(
    uui.Text,
    (props) => {
        return ({
            color: props.color ?? 'gray80',
        });
    },
);
