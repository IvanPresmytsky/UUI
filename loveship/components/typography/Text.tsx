import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type TextColors = 'sky' | 'grass' | 'sun' | 'fire' | 'white' | 'night50' | 'night300' | 'night400' | 'night500' | 'night600' | 'night700' | 'night800' | 'night900' | uui.TextProps['color'];
type TextSize = 'none' | '18' | '24' | '30' | '36' | '42' | '48';

interface TextMods {
    /**
     * Defines component color.
     * @default 'night700'
     */
    color?: TextColors;
    /**
     * Defines text size
     * @default '36'
     */
    size?: TextSize;
    /** Defines text line-height */
    lineHeight?: '12' | '18' | '24' | '30';
    /** Defines text font-size */
    fontSize?: '10' | '12' | '14' | '16' | '18' | '24';
}

/** Represents the properties of the Text component. */
export interface TextProps extends uui.TextCoreProps, TextMods {}

export const Text = createSkinComponent<uui.TextProps, TextProps>(
    uui.Text,
    (props) => {
        return ({
            color: props.color ?? 'night700',
        });
    },
);
