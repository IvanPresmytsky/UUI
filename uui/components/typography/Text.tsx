import { withMods } from '@epam/uui-core';
import { getTextClasses, TextSettings } from '../../helpers';
import { Text as uuiText, TextProps as UuiTextProps } from '@epam/uui-components';
import css from './Text.module.scss';

type TextColor = 'info' | 'warning' | 'error' | 'success' | 'brand' | 'primary' | 'secondary' | 'disabled' | 'white';
type TextSize = 'none' | '18' | '24' | '30' | '36' | '42' | '48';
type TextFontStyle = 'normal' | 'italic';
type TextFontWeight = '200' | '300' | '400' | '600' | '700' | '900';

interface TextMods extends TextSettings {
    color?: TextColor;
    fontWeight?: TextFontWeight;
    fontStyle?: TextFontStyle;
    size?: TextSize;
}

export type TextProps = UuiTextProps & TextMods;

function applyTextMods(mods: TextMods) {
    const textClasses = getTextClasses(
        {
            size: mods.size || '36',
            lineHeight: mods.lineHeight,
            fontSize: mods.fontSize,
        },
        false,
    );

    return [
        css.root,
        'uui-text',
        `uui-color-${mods.color || 'primary'}`,
        `uui-font-weight-${mods.fontWeight || '400'}`,
        `uui-font-style-${mods.fontStyle || 'normal'}`,
        'uui-typography',
    ].concat(textClasses);
}

export const Text = withMods<UuiTextProps, TextMods>(uuiText, applyTextMods);
