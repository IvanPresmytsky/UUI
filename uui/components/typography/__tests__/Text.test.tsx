import React from 'react';
import { Text } from '../Text';
import { renderer } from '@epam/uui-test-utils';

describe('Text', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<Text>Test</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(
                <Text color="primary" size="48" fontSize="24" lineHeight="30" onClick={ jest.fn } rawProps={ { style: { marginRight: '12' } } }>
                    Test
                </Text>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
