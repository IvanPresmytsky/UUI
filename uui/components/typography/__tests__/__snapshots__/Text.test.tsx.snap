// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should be rendered correctly 1`] = `
<div
  className="root uui-text uui-size-36 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography container"
>
  Test
</div>
`;

exports[`Text should be rendered correctly 2`] = `
<div
  className="root uui-text uui-size-48 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography container"
  onClick={[Function]}
  style={
    Object {
      "--uui-text-font-size": "24px",
      "--uui-text-line-height": "30px",
    }
  }
>
  Test
</div>
`;
